### PR TITLE
SSH Remote support for remote linux

### LINKED ISSUE #15: [Bug] Windows terminal with remote linux machine via SSH
There is three bugs with remote ssh workspaces in vscode-windows-terminal extension:
 - the hostname obtained from the workspace path sometimes is not the real hostname of the remote machine.
- the extension is using the windows username in the ssh communication even with linux and it should  be using  the predefined username in ms-vscode-remote.remote-ssh configuration file. 
 - wt.exe does not accept ";" in ssh commands (ex wt.exe ssh ricardo@192.168.1.180 "cd ~;  exec ...") it must be a bug with windows terminal.

P.S. I have a fixed this bugs and will submit them tomorrow as a pull request.

### PR DESCRIPTION
**Pull request by ricardosantos9521**
Implement and test ssh remote for remote linux:
 - obtain hostname and username (with the host provided by the uri) for accessing linux remote machine through the config file of ms-vscode-remote.remote-ssh.
 - fix command given to "wt.exe ssh"

Closes #15

### CODE REVIEW

File: src/ssh.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { readFile } from "fs";
+import * as vscode from "vscode";
+const SSHConfig = require("ssh-config");
```
Comment (Tyriar) [inline src/ssh.ts:3]:
```suggestion
import { readFile } from 'fs';
import * as vscode from 'vscode';
import * as sshConfig from 'ssh-config';
```

File: src/ssh.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  config: Array<IParam>;
+}
+
+export function resolveSSHHostName(host: string) {
+  return new Promise<any | undefined>(function (resolve, reject) {
+    // get path from remote-ssh-extension config file
+    const remoteSettingsPath: string | undefined = vscode.workspace
+      .getConfiguration("remote.SSH")
+      .get("configFile");
+    if (remoteSettingsPath) {
```
Comment (Tyriar) [inline src/ssh.ts:17]: Could you invert this condition to reduce nesting?

File: src/ssh.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      readFile(remoteSettingsPath, function (err, contents) {
+        if (!err) {
```
Comment (Tyriar) [inline src/ssh.ts:19]: Can you invert this if too?

File: src/ssh.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export function resolveSSHHostName(host: string) {
+  return new Promise<any | undefined>(function (resolve, reject) {
+    // get path from remote-ssh-extension config file
+    const remoteSettingsPath: string | undefined = vscode.workspace
+      .getConfiguration("remote.SSH")
+      .get("configFile");
+    if (remoteSettingsPath) {
+      readFile(remoteSettingsPath, function (err, contents) {
```
Comment (Tyriar) [inline src/ssh.ts:18]: Ideally this would use `promisify` like in the other files.

File: src/ssh.ts (hunk 1 above)
Comment (ricardosantos9521) [reply src/ssh.ts:3]: import * as sshConfig from 'ssh-config'; requires @types/ssh-config which i didn't find in npm or a file .d.ts with a declaration should i added it?

Comment (Tyriar) [reply src/ssh.ts:3]: Oh OK, leave it as is for now then

**Pull request merged**